builtin/providers/azurerm/resource_arm_managed_image_test.go

```diff
@@ -0,0 +1,345 @@
+}
+
+resource "azurerm_network_interface" "test" {
+    name = "acctni-%d"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+
+    ip_configuration {
+    	name = "testconfiguration1"
+    	subnet_id = "${azurerm_subnet.test.id}"
+    	private_ip_address_allocation = "dynamic"
+    }
+}
+
+resource "azurerm_managed_image" "test" {
+    name = "accteste"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+	os_disk {
+		os_type = "Linux"
+		os_state = "Generalized"
+		blob_uri = "https://terraformdev.blob.core.windows.net/packerimages/ubuntu_plain.vhd"
+ 	    size_gb = 30
+	}
+
+    tags {
+        environment = "acctest"
+        cost-center = "ops"
+    }
+}
+
+resource "azurerm_virtual_machine" "test" {
+    name = "acctvm"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    network_interface_ids = ["${azurerm_network_interface.test.id}"]
+    vm_size = "Standard_D1_v2"
+
+    storage_image_reference {
+		image_id = "${azurerm_managed_image.test.id}"
+    }
+
+    storage_os_disk {
+        name = "myosdisk1"
+        caching = "ReadWrite"
+        create_option = "FromImage"
+    }
+	
+    os_profile {
+		computer_name = "mdcustomimagetest"
+		admin_username = "testadmin"
+		admin_password = "Password1234!"
+    }
+
+    os_profile_linux_config {
+		disable_password_authentication = false
+    }
+
+    tags {
+    	environment = "Production"
+    	cost-center = "Ops"
+    }
+}
+`
+
+var testAccAzureRMManagedImage_customImage_fromVM = `
+resource "azurerm_resource_group" "test" {
+    name = "acctestRG-%d"
+    location = "West Central US"
+}
+
+resource "azurerm_managed_image" "testdestination" {
+    name = "acctestdest-%d"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+source_virtual_machine_id = "/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/Microsoft.Compute/virtualMachines/{vm_name}"    
+	tags {
+        environment = "acctest"
+        cost-center = "ops"
+    }
+}
+
+resource "azurerm_virtual_network" "test" {
+    name = "acctvn-%d"
+    address_space = ["10.0.0.0/16"]
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_subnet" "test" {
+    name = "acctsub-%d"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    virtual_network_name = "${azurerm_virtual_network.test.name}"
+    address_prefix = "10.0.2.0/24"
+}
+
+resource "azurerm_network_interface" "testdestination" {
+    name = "acctnicdest-%d"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+
+    ip_configuration {
+    	name = "testconfiguration2"
+    	subnet_id = "${azurerm_subnet.test.id}"
+    	private_ip_address_allocation = "dynamic"
+    }
+}
+
+resource "azurerm_virtual_machine" "testdestination" {
```

Comment:
We need an example of a VM using a custom image in either the VM docs or the image docs